Test Fixes

harmonymjb opened the pull request:
This PR attempts to get our tests working again.

It required updates based on other upstream changes with Rails 7, Solidus Ruby support versions, and Ransack 4.

Commit:
Remove circleCI config that uses ruby 3.0

This is no longer available for any supported solidus version

Commit:
Define ransackable_ methods on ::Translation models

After ransack v4, we need to declare these methods on the Translation
model if they are ransackable associations.

https://github.com/activeadmin/activeadmin/discussions/8033#discussioncomment-6564120

Commit:
Change tests to use query param locale instead of route

Routing translations is not directly supported by solidus globalize
and the upstream dependency doesn not support rails 7. In order to
test against newer versions of solidus, these tests are changed to use
the query param instead.

Commit:
Remove routing-filter

This gem was only utilized in the test sandbox, but users are expected
to configure locale routing themselves. Since this gem does not
support rails 7.

https://github.com/svenfuchs/routing-filter/pull/87

jarednorman reviewed (commented):
Looking good, but still seeing some test failures.

harmonymjb commented:
> Looking good, but still seeing some test failures.

It's not entirely clear to me, but it seems this commit may have broken running our tests against `main`:
https://github.com/solidusio/solidus/commit/f2a27ae23bb31ea02083cd7358b689b94bce60cf
The error is just in `create(:shipping_method)` and the entire trace is outside solidus_globalize.

It looks like in combination with the globalize update from @swamp09 , the tests pass, so I've cherry picked their commit here.


Update:

Looks like if we have Globalize 6, the rails 6.1 override for serialization is used:

https://github.com/globalize/globalize/blob/v6.3.0/lib/patches/active_record/rails6_1/serialization.rb
https://github.com/globalize/globalize/blob/v6.3.0/lib/patches/active_record/serialization.rb#L5-L6

whereas in Globalize 7 on Rails 7.1:

https://github.com/globalize/globalize/blob/v7.0.0/lib/patches/active_record/serialization.rb#L3-L4
https://github.com/globalize/globalize/blob/v7.0.0/lib/patches/active_record/rails7_1/serialization.rb

Updating to Globalize 7 will be necessary to support the commit in Solidus, since the Globalize 6 rails override doesn't handle options in the same way.

jarednorman commented:
How much work is that?

Commit:
Override serialize to support Rails 7.1 with Globalize 6

Globalize 6 provides an override for serialize that only suits the
deprecated signature.

This override is applied only if Rails is >= 7.1 and Globalize is < 7
Since globalize 6 does not support Rails 7.2, no handling of Globalize
6 with Rails 7.2 is needed.

The override here is from Globalize v7.0.0 and will be modified in the
next commit.

Commit:
Always pass a class_name_or_coder param to super

The Globalize 6 override that exists in the ancestor chain requires
this value to be passed from :type.

harmonymjb commented:
> How much work is that?

https://github.com/solidusio-contrib/solidus_globalize/compare/0ecf090ac24738b0173f9dd612d90e34b83a6ea1..ea884d5553ac314c089401e5c886a21ee68cd028

I'm not sure this is an awesome solution, but something like this seems to be needed for Globalize 6 with Rails 7.1
